lib/spells/AdventureSpellMechanics.cpp

```diff
@@ -397,15 +417,15 @@ ESpellCastResult TownPortalMechanics::applyAdventureEffects(SpellCastEnvironment
 			return ESpellCastResult::ERROR;
 		}
 
-		const auto relations = env->getCb()->getPlayerRelations(destination->tempOwner, parameters.caster->tempOwner);
+		const auto relations = env->getCb()->getPlayerRelations(destination->tempOwner, parameters.caster->getCasterOwner());
 
 		if(relations == PlayerRelations::ENEMIES)
 		{
 			env->complain("Can't teleport to enemy!");
 			return ESpellCastResult::ERROR;
 		}
 
-		if(static_cast<int>(parameters.caster->movement) < moveCost)
+		if(static_cast<int>(parameters.caster->getHeroCaster()->movement) < moveCost)
```

Comment:
- Will "Town Gate" work same as in Hota if we're at 0 MP when entering building?
- Does "Town Gate" takes MP to begin with?

Reply:
0 points are not allowed (bug?). It doesn't take any movement  points.

Reply:
needs fix, this object is needed for pvp

Reply:
Fixed